packages/git/src/browser/git-scm-provider.spec.ts

```diff
@@ -0,0 +1,102 @@
+import { ScmInput } from '@theia/scm/lib/browser/scm-input';
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import { Git, GitFileStatus, Repository } from '../common';
+import { DugiteGit } from '../node/dugite-git';
+import { GitErrorHandler } from './git-error-handler';
+import { GitPreferences } from './git-preferences';
+import { GitScmProvider, GitScmProviderOptions } from './git-scm-provider';
+
+disableJSDOM();
+
+describe('GitScmProvider', () => {
+    let testContainer: Container;
+    let mockEditorManager: EditorManager;
+    let mockGitErrorHandler: GitErrorHandler;
+    let mockFileService: FileService;
+    let mockGit: Git;
+    let mockCommandService: CommandService;
+    let mockLabelProvider: LabelProvider;
+    let gitScmProvider: GitScmProvider;
+    const repository: Repository = {
+        localUri: 'file:///home/repoA'
+    };
+
+    before(() => {
+        disableJSDOM = enableJSDOM();
+    });
+    after(() => {
+        disableJSDOM();
+    });
+
+    beforeEach(() => {
+        mockEditorManager = sinon.createStubInstance(EditorManager);
+        mockGitErrorHandler = sinon.createStubInstance(GitErrorHandler);
+        mockFileService = sinon.createStubInstance(FileService);
+        mockGit = sinon.createStubInstance(DugiteGit);
+        mockCommandService = {} as CommandService;
+        mockLabelProvider = sinon.createStubInstance(LabelProvider);
+
+        testContainer = new Container();
+        testContainer.bind(EditorManager).toConstantValue(mockEditorManager);
+        testContainer.bind(GitErrorHandler).toConstantValue(mockGitErrorHandler);
+        testContainer.bind(FileService).toConstantValue(mockFileService);
+        testContainer.bind(Git).toConstantValue(mockGit);
+        testContainer.bind(CommandService).toConstantValue(mockCommandService);
+        testContainer.bind(LabelProvider).toConstantValue(mockLabelProvider);
+        testContainer.bind(GitPreferences).toConstantValue({ onPreferenceChanged: () => Disposable.NULL });
+        testContainer.bind(GitScmProviderOptions).toConstantValue({
+            repository
+        } as GitScmProviderOptions);
+        testContainer.bind(GitScmProvider).toSelf();
+        gitScmProvider = testContainer.get(GitScmProvider);
+        gitScmProvider.input = sinon.createStubInstance(ScmInput);
+    });
+
+    it('should unstage all the changes', async () => {
+        const uris = [
+            'file:///home/repoA/test1.txt',
+            'file:///home/repoA/test2.txt'
+        ];
+        gitScmProvider.setStatus({
+            changes: uris.map(uri => ({
+                status: GitFileStatus.New,
+                uri,
+                staged: true
+            })),
+            exists: true
+        });
+        expect(gitScmProvider.stagedChanges.length).to.eq(2);
+
+        await gitScmProvider.unstageAll();
+
+        expect((mockGit.unstage as sinon.SinonStub).calledOnceWith(repository, uris, { reset: 'index' })).to.be.true;
```

Comment:
@Hanksha I'm not sure this test is valuable as is? It is duplicating what is in `git-scm-provider` and expecting it to be called, but we do not check if the result of the call is proper (which is what I expect the test to determine).